File: problems/HackerRank/L2/-pylons.ts
function pylons(k: number, arr: number[]): number {
  // k = 3
  // arr = [0,1,1,1,0,0,0]
  // -1
  // k = 2
  // arr = [0,1,1,1,1,0]
  // 2
  // k = 2
  // arr = [0,1,1,0,0,0,0,1,1]
  // -1
  // k = 3
  // arr = [0,1,1,1,0,0,0]
  // -1
  // k = 3
  // arr = [0,1,1,1,0,0]
  // 2

  // let i = 0;
  // let count = 0;

  // while (i < arr.length) {
  //   let j = i + k - 1;
  //   while (j > i - k) {
  //     if (arr[j] == 1) {
  //       count++;
  //       i = j + k;
  //       break;
  //     }
  //     j--;
  //     if (j == i - k) j--;
  //   }
  //   if (j < i - k) return -1;
  // }
  return 0;
}

File: problems/HackerRank/week7/pylons.py
def pylons(k, arr):
    
    n = len(arr)
    i = 0
    ans = 0
    
    while i < n:
        rightmost = min(n - 1, i + (k - 1))
        leftmost = max(0, i - (k - 1))
        
        position = -1
        for j in range(rightmost, leftmost -1, -1):
            if arr[j] == 1:
                position = j
                break
        
        if position == -1:
            return -1
        
        ans += 1
        
        i = position + k
        
    return ans